src/backend/joanie/core/serializers.py

```diff
@@ -265,18 +265,17 @@ def get_orders(self, instance):
         try:
             username = self.context["username"]
             orders = (
-                models.Order.objects.filter(
-                    owner__username=username,
-                    course=instance,
-                    state__in=[
-                        enums.ORDER_STATE_FAILED,
-                        enums.ORDER_STATE_FINISHED,
-                        enums.ORDER_STATE_PAID,
-                        enums.ORDER_STATE_PENDING,
-                    ],
+                order
+                for order in (
+                    models.Order.objects.filter(
+                        owner__username=username,
+                        course=instance,
+                        is_canceled=False,
+                    )
+                    .select_related("product")
+                    .prefetch_related("enrollments__course_run")
                 )
-                .select_related("product")
-                .prefetch_related("enrollments__course_run")
+                if order.state == enums.ORDER_STATE_VALIDATED
```

Comment:
why not add a parameter to your filer : state=enums.ORDER_STATE_VALIDATED
Apart from the technical query, I'm not sure why we only want the validated orders here.

Reply:
Because in `CourseSerializer` we want only want to know if user already owns a related product. To know that, from frontend we check if a product has a corresponding validated order.

Reply:
Oh yes, good idea ! 👍